Jellyfin.Server.Implementations/AudioWaveform/AudioWaveformManager.cs

```diff
@@ -0,0 +1,170 @@
+            ? Path.Combine(item.ContainingFolderPath, saveFileName)
+            : Path.Combine(item.GetInternalMetadataPath(), saveFileName);
+        string outputPathCsv = Path.Combine(tempDir, saveFileName.Replace("json", "csv", StringComparison.Ordinal));
+
+        if (!File.Exists(outputPathJson))
+        {
+            int sampleRate = 0;
+            int fileSampleRate = item.GetMediaStreams()[0].SampleRate ?? 44100;
+            // This value sets the length of a frame, the amount of measurements per second.
+            int samplesPerSecond = 2;
+            sampleRate = Math.Max(fileSampleRate / samplesPerSecond, 1);
+
+            // escape double quotes
+            string itemPathEscaped = item.Path.Replace("\"", "\\\"", StringComparison.Ordinal);
+            string outputPathCsvEscaped = outputPathCsv.Replace("\"", "\\\"", StringComparison.Ordinal);
+
+            var process = Process.Start("ffprobe", $"-v error -f lavfi -i \"amovie={itemPathEscaped},asetnsamples={sampleRate},astats=metadata=1:reset=1\" " + $"-show_entries frame_tags=lavfi.astats.Overall.RMS_peak -of csv=p=0 -o \"{outputPathCsvEscaped}\"");
+
+            var waitTask = process.WaitForExitAsync();
+            int timeout = 10; // seconds, this is quite an arbitrary value
+            var timeoutTask = Task.Delay(TimeSpan.FromSeconds(timeout));
+            var completedTask = await Task.WhenAny(waitTask, timeoutTask).ConfigureAwait(false);
+
+            if (completedTask == timeoutTask)
+            {
+                process.Kill();
+                await process.WaitForExitAsync().ConfigureAwait(false); // Ensure process is fully terminated
+                throw new TimeoutException("ffprobe took too long and was terminated.");
+            }
+
+            var csvLines = await File.ReadAllLinesAsync(outputPathCsv).ConfigureAwait(false);
+            var samples = csvLines
+                    .Select(v =>
+                {
+                    if (string.Equals(v.Trim(), "-inf", StringComparison.Ordinal))
+                    {
+                        return 0.0;  // sometimes ffprobe returns "-inf" instead of a number
+                    }
+                    else
+                    {
+                        return double.TryParse(v, NumberStyles.Float, CultureInfo.InvariantCulture, out double parsedValue)
+                            ? Math.Pow(2, parsedValue / 6)
+                            : 0.0;
+                    }
+                })
+                .ToArray();
+
+            var data = new Dictionary<string, object>
+            {
+                { "samples", samples },
+                { "sampleRate", sampleRate }
+            };
+
+            var json = JsonSerializer.Serialize(data);
+            await File.WriteAllTextAsync(outputPathJson, json).ConfigureAwait(false);
+            File.Delete(outputPathCsv);
+        }
+
+        var fileStream = new FileStream(outputPathJson, FileMode.Open, FileAccess.Read, FileShare.Read);
```

Comment:
move the creation of the FileStream as late as possible. If something crashes in the SaveAudioWaveform method, this might lead to memory leaks